File: 2017-10-06 Cerquem!/p5.py
# Traducció P90260

from jutge import read


# Algun llest es va deixar de posar constants en Python! 😢
# La convenció és que les variables en MAJÚSCULES són constants 
# i no s'haurien de tocar (però no està prohibit).
#
# Com a mínim, les variables en Python poden tenir accents, en C++ no.


ALÇADA_PICA = 3143
ÚLTIM_ELEMENT = 0


def pic_guapo(x, y, z):
    return x < y and y > z and y > ALÇADA_PICA


def main():
    trobat = False
    a, b, c = read(int, int, int)
    while not trobat and c != ÚLTIM_ELEMENT:
        if pic_guapo(a, b, c):
            trobat = True
        else:
            a, b, c = b, c, read(int)
    print("SI" if trobat else "NO")       # operador condicional en Python com el ?: de C++.


main()

File: 2017-10-06 Cerquem!/p5.cc
// Trobar si hi ha un pic més alt que la Pica d'Estats (P90260)
// => Cerca

#include <iostream>
using namespace std;


// definim constants (es solen escriure en MAJÚSCULES) per no tenir valors màgics al mig del codi
const int ALCADA_PICA = 3143;   
const int ULTIM_ELEMENT = 0;


bool pic_guapo(int x, int y, int z) {
    return x < y and y > z and y > ALCADA_PICA;
}


int main() {
    bool trobat = false;
    int a, b, c;
    cin >> a >> b >> c;
    while (not trobat and c != ULTIM_ELEMENT) {
        if (pic_guapo(a, b, c)) {
            trobat = true;
        } else {
            a = b;                
            b = c;
            cin >> c;
        }
    }
    cout << (trobat ? "SI" : "NO") << endl;
}